File: gdscript/src/main/java/gdscript/psi/impl/GdFlowStImpl.java
package gdscript.psi.impl;

import org.jetbrains.annotations.*;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import com.intellij.psi.util.PsiTreeUtil;
import gdscript.psi.*;

public class GdFlowStImpl extends GdStmtImpl implements GdFlowSt {

  public GdFlowStImpl(@NotNull ASTNode node) {
    super(node);
  }

  @Override
  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitFlowSt(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

  @Override
  @NotNull
  public GdEndStmt getEndStmt() {
    return notNullChild(PsiTreeUtil.getChildOfType(this, GdEndStmt.class));
  }

  @Override
  @Nullable
  public GdExpr getExpr() {
    return PsiTreeUtil.getChildOfType(this, GdExpr.class);
  }

  @Override
  @NotNull
  public String getType() {
    return GdPsiUtils.getType(this);
  }

}

File: gdscript/src/main/kotlin/gdscript/psi/impl/GdFlowStImpl.kt
package gdscript.psi.impl

import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElementVisitor
import com.intellij.psi.util.PsiTreeUtil
import gdscript.psi.GdEndStmt
import gdscript.psi.GdExpr
import gdscript.psi.GdFlowSt
import gdscript.psi.GdPsiUtils.getType
import gdscript.psi.GdVisitor

class GdFlowStImpl(node: ASTNode) : GdStmtImpl(node), GdFlowSt {
    override fun accept(visitor: GdVisitor) {
        visitor.visitFlowSt(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }

    override val endStmt: GdEndStmt
        get() = notNullChild(PsiTreeUtil.getChildOfType(this, GdEndStmt::class.java))

    override val expr: GdExpr?
        get() = PsiTreeUtil.getChildOfType(this, GdExpr::class.java)

    override val type: String
        get() = getType(this)
}
